clippy_lints/src/methods/unnecessary_min_or_max.rs

```diff
@@ -0,0 +1,92 @@
+use std::cmp::Ordering;
+
+use super::UNNECESSARY_MIN_OR_MAX;
+use clippy_utils::diagnostics::span_lint_and_sugg;
+
+use clippy_utils::consts::{constant, constant_with_source, Constant, ConstantSource, FullInt};
+use clippy_utils::source::snippet;
+
+use rustc_errors::Applicability;
+use rustc_hir::Expr;
+use rustc_lint::LateContext;
+use rustc_middle::ty;
+use rustc_span::Span;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx Expr<'_>,
+    name: &str,
+    recv: &'tcx Expr<'_>,
+    arg: &'tcx Expr<'_>,
+) {
+    let typeck_results = cx.typeck_results();
+    if let (
+        Some((left, ConstantSource::Local | ConstantSource::CoreConstant)),
+        Some((right, ConstantSource::Local | ConstantSource::CoreConstant)),
+    ) = (
+        constant_with_source(cx, typeck_results, recv),
+        constant_with_source(cx, typeck_results, arg),
+    ) {
```

Comment:
Let's go with `if let Some(..) == constant_with_source(..) && let Some(..) = constant_with_source(..)` so we don't have to evaluate `arg` as a constant if `recv` isn't one